src/abstractdataframe/abstractdataframe.jl

```diff
@@ -233,8 +233,30 @@ Base.ndims(::AbstractDataFrame) = 2
 ##
 ##############################################################################
 
-Base.similar(df::AbstractDataFrame, dims::Int) =
-    DataFrame(Any[similar_missing(x, dims) for x in columns(df)], copy(index(df)))
+"""
+    similar(df::DataFrame[, rows::Int])
+
+Create a new DataFrame with the same column names and column element-types
+as `df`. An optional second argument can be provided to request a number of rows
+that is different than the number of rows present in `df`.
+"""
+function Base.similar(df::AbstractDataFrame, rows::Integer = size(df, 1))
+    @assert rows >= 0 "the number of rows must be positive"
+    DataFrame(Any[similar(x, rows) for x in columns(df)], copy(index(df)))
+end
+
+"""
+    similar_missing(df::DataFrame[, rows::Int])
```

Comment:
I'd prefer if we avoided adding this function until we're completely certain it's needed. People can use `allowmissing!`, and hopefully the compiler will be able to make this efficient at some point.